internal/datanode/compactor.go

```diff
@@ -389,20 +454,21 @@ func (t *compactionTask) merge(
 			}
 		}
 	}
-	if currentRows != 0 {
-		uploadInsertStart := time.Now()
-		inPaths, statsPaths, err := t.uploadSingleInsertLog(ctxTimeout, targetSegID, partID, meta, fID2Content, fID2Type)
+
+	// upload stats log and remain insert rows
+	if numRows != 0 || currentRows != 0 {
```

Comment:
do we need || currentRows != 0 ?

Reply:
or if it is better to do
if currentRows != 0 {
uploadSingleInsertLog()
}
if numRows != 0 {
uploadStatsLog()
}

Reply:
> 



> do we need || currentRows != 0 ?

Because numRows was a  estimated value(row num before merge) use for create bloom filter, but currnentRows was the true rows after we review all data.
I could only use currentRows, because merge only reduce num rows, so currentRows<=numRows.

Reply:
> or if it is better to do if currentRows != 0 { uploadSingleInsertLog() } if numRows != 0 { uploadStatsLog() }

Merge for reduce io pool and rpc use. And two rows both used by stats log, numRows was rows of statslog BF  create, currentRows was true insert rows.